### ISSUE TITLE
Invalid non-numeric limit passed to limitResult()

### ISSUE THREAD

**Issue by Nikerabbit** [SemanticMediaWiki/SemanticMediaWiki#2479]
### Setup and configuration

- SMW version: 2.5.2
- MW version: 1.30.0
- PHP version: 3.9.12 (hhvm)
- DB (MySQL etc.): 10.0.29

### Issue

Produces a [stack trace](https://www.semantic-mediawiki.org/wiki/Help:Identifying_bugs) and outputs:

```
RuntimeException from line 229 of ////extensions/SemanticMediaWiki/src/MediaWiki/Database.php: Invalid non-numeric limit passed to limitResult()

#0 ////includes/libs/rdbms/database/Database.php(1381): Wikimedia\Rdbms\Database->limitResult()
#1 ////includes/libs/rdbms/database/Database.php(1327): Wikimedia\Rdbms\Database->selectSQLText()
#2 ////extensions/SemanticMediaWiki/src/MediaWiki/Database.php(224): Wikimedia\Rdbms\Database->select()
#3 ////extensions/SemanticMediaWiki/includes/storage/SQLStore/SMW_SQLStore3_Readers.php(504): SMW\MediaWiki\Database->select()
#4 ////extensions/SemanticMediaWiki/includes/storage/SQLStore/SMW_SQLStore3_Readers.php(592): SMWSQLStore3Readers->getPropertySubjects()
#5 ////extensions/SemanticMediaWiki/src/SQLStore/EntityStore/DirectEntityLookup.php(86): SMWSQLStore3Readers->getAllPropertySubjects()
#6 ////extensions/SemanticMediaWiki/includes/storage/SQLStore/SMW_SQLStore3.php(225): SMW\SQLStore\EntityStore\DirectEntityLookup->getAllPropertySubjects()
#7 ////extensions/SemanticMediaWiki/includes/articlepages/SMW_PropertyPage.php(246): SMWSQLStore3->getAllPropertySubjects()
#8 ////extensions/SemanticMediaWiki/includes/articlepages/SMW_PropertyPage.php(80): SMWPropertyPage->getPropertyValueList()
#9 ////extensions/SemanticMediaWiki/includes/articlepages/SMW_OrderedListPage.php(186): SMWPropertyPage->getHtml()
#10 ////extensions/SemanticMediaWiki/includes/articlepages/SMW_OrderedListPage.php(94): SMWOrderedListPage->showList()
#11 ////includes/actions/ViewAction.php(68): SMWOrderedListPage->view()
#12 ////includes/MediaWiki.php(499): ViewAction->show()
#13 ////includes/MediaWiki.php(293): MediaWiki->performAction()
#14 ////includes/MediaWiki.php(862): MediaWiki->performRequest()
#15 ////includes/MediaWiki.php(523): MediaWiki->main()
#16 ////index.php(43): MediaWiki->run()
#17 {main}
#0 ////extensions/SemanticMediaWiki/includes/storage/SQLStore/SMW_SQLStore3_Readers.php(504): SMW\MediaWiki\Database->select(string, string, string, string, array)
#1 ////extensions/SemanticMediaWiki/includes/storage/SQLStore/SMW_SQLStore3_Readers.php(592): SMWSQLStore3Readers->getPropertySubjects(SMW\DIProperty, NULL, SMW\RequestOptions)
#2 ////extensions/SemanticMediaWiki/src/SQLStore/EntityStore/DirectEntityLookup.php(86): SMWSQLStore3Readers->getAllPropertySubjects(SMW\DIProperty, SMW\RequestOptions)
#3 ////extensions/SemanticMediaWiki/includes/storage/SQLStore/SMW_SQLStore3.php(225): SMW\SQLStore\EntityStore\DirectEntityLookup->getAllPropertySubjects(SMW\DIProperty, SMW\RequestOptions)
#4 ////extensions/SemanticMediaWiki/includes/articlepages/SMW_PropertyPage.php(246): SMWSQLStore3->getAllPropertySubjects(SMW\DIProperty, SMW\RequestOptions)
#5 ////extensions/SemanticMediaWiki/includes/articlepages/SMW_PropertyPage.php(80): SMWPropertyPage->getPropertyValueList()
#6 ////extensions/SemanticMediaWiki/includes/articlepages/SMW_OrderedListPage.php(186): SMWPropertyPage->getHtml()
#7 ////extensions/SemanticMediaWiki/includes/articlepages/SMW_OrderedListPage.php(94): SMWOrderedListPage->showList()
#8 ////includes/actions/ViewAction.php(68): SMWOrderedListPage->view()
#9 ////includes/MediaWiki.php(499): ViewAction->show()
#10 ////includes/MediaWiki.php(293): MediaWiki->performAction(SMWPropertyPage, Title)
#11 ////includes/MediaWiki.php(862): MediaWiki->performRequest()
#12 ////includes/MediaWiki.php(523): MediaWiki->main()
#13 ////index.php(43): MediaWiki->run()
#14 {main}
```

Steps to reproduce the observation (recommendation is to use the [sandbox](https://sandbox.semantic-mediawiki.org)):
https://sandbox.semantic-mediawiki.org/w/index.php?title=Attribut:Step&limit=20ave&offset=0&value=&from=&until=

NB: I noticed this due to exploitation attempts that flooded the logs.

**Comment by kghbln** [comment]
Thanks for reporting!

**Comment by mwjames** [comment]
> RuntimeException from line 229 of ////extensions/SemanticMediaWiki/src/MediaWiki/Database.php: Invalid non-numeric limit passed to limitResult()

Fixed with #2481.
